### DISCUSSION TITLE
Hilfe mit Config

### DISCUSSION THREAD

**Discussion by mozzi75** [jomjol/AI-on-the-edge-device#2353] [Q&A]
Hallo zusammen.
Erstmal super Projekt.
Ich bin da leider nur etwas daneben mit den Einstellungen.
Die ROIs werden richtig erkannt Mittlerweile, aber die Umrechnung bekomme ich nicht hin.
Meine Config:
```
[TakeImage]
;RawImagesLocation = /log/source
WaitBeforeTakingPicture = 5
;RawImagesRetention = 15
Demo = false
Brightness = 2
Contrast = 0
Saturation = 0
LEDIntensity = 8
ImageQuality = 12
ImageSize = VGA
FixedExposure = false

[Alignment]
InitialRotate = 90
InitialMirror = false
SearchFieldX = 20
SearchFieldY = 20
AlignmentAlgo = default
FlipImageSize = false
/config/ref0.jpg 315 126
/config/ref1.jpg 374 166

[Digits]
Model = /config/dig-class11_1600_s2.tflite
CNNGoodThreshold = 0.5
;ROIImagesLocation = /log/digit
;ROIImagesRetention = 3
main.dig1 230 157 24 43 true
main.dig2 257 157 24 43 0
main.dig3 284 157 24 43 0
main.dig4 311 157 24 43 0
main.dig5 338 157 24 43 0

[Analog]
Model = /config/ana-cont_1105_s2_q.tflite
;ROIImagesLocation = /log/analog
;ROIImagesRetention = 3
main.ana3 215 332 69 69 false
main.ana4 154 270 69 69 false
main.ana2 300 324 69 69 false
main.ana1 348 252 69 69 false

[PostProcessing]
main.DecimalShift = 3
main.AnalogDigitalTransitionStart = 9.2
PreValueUse = true
PreValueAgeStartup = 720
main.AllowNegativeRates = false
main.MaxRateValue = 0.5
main.MaxRateType = AbsoluteChange
main.ExtendedResolution = false
main.IgnoreLeadingNaN = false
ErrorMessage = true
CheckDigitIncreaseConsistency = false
```

Ich habe hier einen DIEHL Wasserzähler mit 5 Digitalen Stellen und 4 analogen Messzeigern.
Sollten noch Infos gebraucht werden bitte um kurzes Feedback.
Und egal was ich umstelle kommt entweder: Rate to high oder rate to low.
Das ist der Zähler der verbaut ist:
(https://www.diehl.com/metering/de/produkte-loesungen/produkte-services/wasserzaehler/corona-mw-at/)

**Comment by caco3** [comment]
Schalt mal die Checks aus und lass ihn ein paar Tage laufen.

Und dann schau dir den Graph an und die Max rate:
![grafik](https://user-images.githubusercontent.com/1783586/234994419-a38dbf8d-0b2a-4949-a577-a2728f8f5f4a.png)

Bei mir sind das somit max. 6 Liter pro Interval (2 Minuten).  ich hatte abe rauch schon mal 12 Liter gesehen.
Daher setze ich `MaxRateValue` auf z.B. 50 Liter (0.05).

**Reply by mozzi75** [reply]
Danke.
Werte sind mal da.
<img width="640" alt="image" src="https://user-images.githubusercontent.com/49696696/235149619-af06b012-8905-4fc6-a731-fe1f767f81e9.png">

**Reply by mozzi75** [reply]
<img width="480" alt="image" src="https://user-images.githubusercontent.com/49696696/235149869-508de20a-ce2b-4f82-8c64-48ebcabbd885.png">

**Reply by caco3** [reply]
Das sieht doch gut aus.

Kannst Du etwas am Kontrast schrauben?

Was hast Du für ein Interval?

**Reply by mozzi75** [reply]
Hab den Kontrast auf 5 gesetzt.
welchen Intervall meinst du?

**Reply by caco3** [reply]
> welchen Intervall meinst du?

https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/?v=b24b7d5#parameter-interval

**Reply by mozzi75** [reply]
Den hab ich standardmässig auf 5 min gelassen

**Comment by ghost** [comment]
Die Reihenfolge deiner analogen ROIs ist falsch. Auf dem Bild ist 109.5533 zu sehen

**Reply by mozzi75** [reply]
lt. Bild sollte sie richtig sein.
Nur komisch das er beim dropdown falsch anzeigt.
<img width="414" alt="image" src="https://user-images.githubusercontent.com/49696696/235232342-a5d0f69c-59fd-47be-b0f5-d315fb216664.png">

**Comment by ghost** [comment]
Leider ist hier die Reihenfolge der ROI entscheidend. Nicht der Name. Am besten einfach in der Config-Datei in die richtige Reihenfolge bringen.

@caco3 : Ich bin ja dafür, dass wir gar keine Namen mehr vergeben lassen, sondern sie generieren. Das hat schon öfter zu Verwirrung geführt.

**Reply by mozzi75** [reply]
Danke. sollte so stimmen.
Stimmt hier war es verdreht.

main.ana1 348 252 69 69 false
main.ana2 300 324 69 69 false
main.ana3 215 332 69 69 false
main.ana4 154 270 69 69 false

**Reply by caco3** [reply]
> Ich bin ja dafür, dass wir gar keine Namen mehr vergeben lassen, sondern sie generieren. Das hat schon öfter zu Verwirrung geführt.

Ja, das wäre vermutlich besser.
